src/utils/mmap_backend.rs

```diff
@@ -142,6 +149,14 @@ impl<W> MmapBackend<W, MmapMut> {
         // Align to multiple of size_of::<W>
         let mmap_len = file_len.align_to(size_of::<W>());
 
+        #[cfg(windows)]
```

Comment:
Didn't we say we don't write to files?

Reply:
We don't really have a choice on this. When we call `map_mut` on Windows the file is extended with random staff to the size of the memory mapping (see [CreateFileMappingW](https://learn.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-createfilemappingw#remarks)). This ensures the file is extended with 0s instead of random bytes

Reply:
Ok. In this case, we should fail if the file is not padded correctly, exactly as in the immutable case. We do not want the length of the file change under any circumstance.